### PR TITLE
Blocks: Normalize RichText value from string (as HTML), children

### PR DESCRIPTION
**Pull request by aduth**
The idea of this PR is to ensure previous templates still work while avoiding leaking the new RichText structure format into the format of the templates.

```php
<?php

/**
 * Plugin Name: Demo CPT
 */

add_action( 'init', function() {
	register_post_type( 'book', [
		'label' => 'Book',
		'show_in_rest' => true,
		'public' => true,
		'show_ui' => true,
		'supports' => [ 'title', 'editor' ],
		'template' => [
			[ 'core/paragraph', [
				'content' => 'Hello <strong>world</strong>!',
			] ],
			[ 'core/paragraph', [
				'content' => [
					'Hello ',
					[
						'type' => 'strong', 
						'props' => [ 'children' => 'world' ]
					],
					'!'
				],
			] ],
			[ 'core/paragraph', [
				'content' => [
					'text' => 'Hello world!',
					'formats' => [
						null,
						null,
						null,
						null,
						null,
						null,
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						null
					],
				],
			] ],
		],
		'template_lock' => 'all',
	] );
} );
```

### CODE REVIEW

**Comment by mcsf** [comment]
Adding to 4.0 as we need a solid compat plan for RichText before releasing its format changes.

**Comment by gziolo** [comment]
Can we prevent the following format from leaking into templates for posts and inner blocks?

```js
			[ 'core/paragraph', [
				'content' => [
					'text' => 'Hello world!',
					'formats' => [
						null,
						null,
						null,
						null,
						null,
						null,
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						[ [ 'type' => 'strong' ] ],
						null
					],
				],
			] ],
```

**Comment by youknowriad** [comment]
I made some updates to this PR. The idea is to only support the "element/children" format for templates (which is what we currently do prior to 4.0) and also updates the code to avoid baking it in `createBlock`. Making it explicit that it's about templates.

Also, I'm fixing the "nested" attributes normalization (query sources) so, something like that would work:

```php
add_action( 'init', function() {
	register_post_type( 'book', [
		'label' => 'Book',
		'show_in_rest' => true,
		'public' => true,
		'show_ui' => true,
		'supports' => [ 'title', 'editor' ],
		'template' => [
			[ 'core/paragraph', [
				'content' => 'Hello first paragraph!',
			] ],
			[ 'core/paragraph', [
				'content' => [
					'Hello ',
					[
						'type' => 'strong',
						'props' => [ 'children' => 'world' ]
					],
					'!'
				],
			] ],

			[ 'core/gallery', [
				'images' => [
					[
						'url' => 'https://picsum.photos/200/300',
						'caption' => [
							'Hello ',
								[
									'type' => 'strong',
									'props' => [ 'children' => 'world' ]
								],
								'!'
						]
					]
				],
			] ],
		],
		'template_lock' => 'all',
	] );
} );
```

File: packages/blocks/src/api/factory.js [hunk 1]
```diff
@@ -69,14 +67,18 @@ export function createBlock( name, blockAttributes = {}, innerBlocks = [] ) {
 			} );
 
 			// Ensure value passed is always an array, which we're expecting in
-			// the RichText component to handle the deprecated value.
-			if ( typeof result[ key ] === 'string' ) {
-				result[ key ] = [ result[ key ] ];
-			} else if ( ! Array.isArray( result[ key ] ) ) {
-				result[ key ] = [];
+			// the RichText getColorObjectByAttributeValuescomponent to handle the deprecated value.
```
Comment (gziolo) [inline packages/blocks/src/api/factory.js]: Is this copy and paste issue `getColorObjectByAttributeValuescomponent`?

Comment (youknowriad) [reply packages/blocks/src/api/factory.js]: hahaha :) sorry about that

**Comment by gziolo** [comment]
Superseded by #10439.

**Pull request closed without merging**